src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.internet.shop.service.impl;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.OrderService;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        return orderDao.completeOrder(shoppingCart);
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        return orderDao.getUserOrders(userId);
+    }
+
+    @Override
+    public Order get(Long id) {
+        return orderDao.get(id).orElseThrow(() ->
+                new IllegalArgumentException(" This " + id + " doesn't exist!"));
```

Comment:
```suggestion
        return orderDao.get(id).get();
```

Reply:
If you try to extract the value of the null element, you'll receive an exception by default.
No need to throw another one instead.

Reply:
ok, fixed thx.